code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/dao/impl/ServiceConsumeMapDaoImpl.java

```diff
@@ -103,6 +107,7 @@ public Instance findOneInstanceForService(long serviceId) {
                 .join(SERVICE_EXPOSE_MAP)
```

Comment:
we should exclude instances in Removing and Removed states here. ServiceExpose map gets removed as a prehandler for instance.remove, but there is still a chance that instance was put to Removed/Removing state and serviceExposeMap is still present.